lib/rucio/common/logging.py

```diff
@@ -160,11 +179,11 @@ def _timestamp_formatter(record_formatter: "LogDataSource", record: "LogRecord")
     yield record_formatter.ecs_fields[0], datetime.datetime.utcfromtimestamp(record.created).isoformat(timespec='milliseconds') + 'Z'
 
 
-def _ecs_field_to_record_attribute(field_name):
+def _ecs_field_to_record_attribute(field_name: Union[ECS_FIELDS, str]) -> Union[LOG_RECORDS, str]:
     """
     Sanitize the path-like field name into a symbol which can be the name of an object attribute.
     """
-    record = ECS_TO_LOG_RECORD_MAP.get(field_name)
+    record = ECS_TO_LOG_RECORD_MAP.get(field_name)  # type: ignore
```

Comment:
The type ignore here is to ignore the error that occurs when a `str` that's not in `ECS_FIELDS` is passed to `_ecs_field_to_record_attribute`, in which case the function returns `field_name.replace('-', '_').replace('.', '_')`.

I personally don't like this logic being in this function, as the function name doesn't say anything about replacing those symbols within a string, and from the function name `_ecs_field_to_record_attribute` you would assume that this only takes in `ECS_FIELDS` and returns `LOG_RECORDS`.

I think it would be a good idea to change the way this works in future.